projects/hslayers/src/components/styles/styler.service.ts

```diff
@@ -751,4 +783,111 @@ export class HsStylerService {
       this.hsLogService.warn('SLD could not be parsed', err);
     }
   }
+
+  /**
+   * Calculate a stop-color value for gradient. Offset depend on number of colors (lenght)
+   * and current index.
+   * @param color RGB(A) color definition
+   * @param linearGradient SVG Linear Gradient element
+   * @param length Offset calculation - length of an array (number of colors)
+   * @param index Index of color
+   */
+  private appendColorToGradient(
+    color: number[],
+    linearGradient: SVGLinearGradientElement,
+    length: number,
+    index: number,
+  ): void {
+    const stop = document.createElementNS('http://www.w3.org/2000/svg', 'stop');
+    const step = 100 / length;
+    const offset = index * step;
+    stop.setAttribute('offset', `${100 - offset}%`);
+    const rgb = `rgb(${color[0]},${color[1]},${color[2]})`;
+    stop.setAttribute('stop-color', rgb);
+
+    linearGradient.appendChild(stop);
+  }
+
+  /**
+   * Generate SVG gradient for selected color map. Used by either
+   * - interpolated layer legend or
+   * - color map select options
+   */
+  generateSVGGradientForColorMap(input: Layer<any> | number[][]) {
+    const forLayer = input instanceof Layer;
+    const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    svg.setAttribute('width', forLayer ? '75%' : '100%');
+    svg.setAttribute('height', '100%');
+
+    const defs = document.createElementNS('http://www.w3.org/2000/svg', 'defs');
+    const linearGradient = document.createElementNS(
+      'http://www.w3.org/2000/svg',
+      'linearGradient',
+    );
+
+    const id = `${+new Date()}-idwGradient-${
+      forLayer
+        ? getLaymanFriendlyLayerName(input.get('name'))
+        : (Math.random() * 1000000).toFixed(0).toString()
+    }`;
+
+    linearGradient.setAttribute('id', id);
+    linearGradient.setAttribute(
+      'gradientTransform',
+      `rotate(${input instanceof Layer ? 90 : 0})`,
+    );
+
+    const rect = document.createElementNS('http://www.w3.org/2000/svg', 'rect');
+    rect.setAttribute('width', forLayer ? '50%' : '100%');
+    rect.setAttribute('height', '100%');
+    rect.setAttribute('fill', `url('#${id}')`);
+
+    //Gradient def generation
+    if (input instanceof Layer) {
+      const source = input.getSource();
+      const arr = Array.from(Array(100).keys());
+      for (const i of arr.filter((e, i) => i % 5 === 5 - 1).reverse()) {
```

Comment:
```suggestion
      for (const i of arr.filter((e, idx) => idx % 5 === 4).reverse()) {
```

`5 - 1` may have sense if you intend to make a variable from 5.
I propose to rename the internal `i` as it is used twice in the same line but in different context.

Reply:
This ones interesting :D I cannot quite figure out how come I ended up with 5-1 :D . Probably was just thinking about making it variable as you suggested or wsa figuring out how to make it work. Will make suggested changes tho, thanks